### PR TITLE
PagerankGraph: add neighbors + score decomposition

### PR DESCRIPTION
**Pull request by decentralion**
This commit adds a `neighbors` method to `PagerankGraph`. This is an
augmented version of `Graph.neighbors`. It returns the base data from
`Graph.neighbors` as well as the score, the edge weights, and the score
contribution. The score contribution basically means how much score was
contributed from the target node by this particular neighbor connection.

When the graph is well-converged, a node's score will be the sum of all
its neighbors' score contributions, as well as the contribution it
received from its synthetic loop edge. So, for completeness sake, I
added another method, `syntheticLoopScoreContribution`, which computes
how much score a node received from its synthetic loop edge. (This value
should usually be close to 0).

You can think of these two methods as providing a replacement for the
`PagerankNodeDecomposition` logic.

Test plan: I've added tests that verify:
- That neighbors returns results consistent with Graph.neighbors
- That neighbors' score contributions are computed correctly
- That neighbors errors if the graph has been modified
- That synthetic score contributions are computed correctly
- That a node's score is the sum of all its contributions

Test plan: Unit tests included. Run `yarn test`.

### CODE REVIEW

File: src/core/pagerankGraph.js [hunk 1]
```diff
@@ -286,6 +299,88 @@ export class PagerankGraph {
...
+      let relevantEdgeWeight = 0;
+      if (edge.src === target) {
+        relevantEdgeWeight += weightedEdge.weight.froWeight;
+      }
+      if (edge.dst === target) {
+        relevantEdgeWeight += weightedEdge.weight.toWeight;
+      }
+      // We normalize this edge weight by the total outWeight for `node`.
+      const normalizedEdgeWeight =
+        relevantEdgeWeight / this.totalOutWeight(node);
```
Comment (BrianLitwin) [inline src/core/pagerankGraph.js]: If you ask for outgoing edges in `options`, do you still want to divide those weights by `totalOutWeight` of `node`?Wouldn't you be dividing `node`'s incoming edge weight by its `totalOutWeight`? And the total incoming edge weight does not equal the total outgoing edge weight? (but total incoming score does equal total outgoing score, doesn't it?? Which sort of makes this difficult to model on a napkin)

So maybe if you are asking for outgoing edges, divide by the `totalOutWeight` of `target`? So if you ask for incoming edges, the `scoredContribution` is `node` 's contribution to `target`, and if you ask for outgoing edges, you get `target`'s contribution to `node` ?

Comment (decentralion) [reply src/core/pagerankGraph.js]: Remember that for the purposes of pagerank/the weighted graph, every edge is actually two edges: one pointing from src to dst, and one pointing from dst to src. So even if we are filtering for outgoing edges (i.e. where target is the src), we still have an edge coming back from `node` to `target`. Then to get the normalized weight of the edge from `node` to `target`, we need to normalize by `node`'s total weight (because the markov chain rule is that the weight of all the connections out of a node need to sum to one).

Basically, the concept of "outgoing edge" at the neighbor API level is orthogonal to the concept of "outgoing connection", since every edge (whether inbound or outbound) corresponds to two connections.

Does that make sense?

Comment (BrianLitwin) [reply src/core/pagerankGraph.js]: I got `froWeight` and `toWeight` mixed up. I was reading the following snippet as accumulating weights from `target` to `node` instead of `node` to `target`: 
```javascript 
 if (edge.src === target) {
        relevantEdgeWeight += weightedEdge.weight.froWeight;
  }
```
